test/e2e/specs/infrastructure/notification__actions.ts

```diff
@@ -0,0 +1,127 @@
+	const comment = DataHelper.getRandomPhrase() + ' notification-actions-spec';
+
+	let notificationsComponent: NotificationsComponent;
+	let restAPIClient: RestAPIClient;
+	let siteID: number;
+	let commentID: number;
+
+	describe( `Leave a comment as ${ commentingUser }`, function () {
+		let postID: number;
+
+		beforeAll( async function () {
+			restAPIClient = new RestAPIClient( SecretsManager.secrets.testAccounts.commentingUser );
+		} );
+
+		it( 'Get the latest post in reader stream', async function () {
+			const response: ReaderResponse = await restAPIClient.getReaderFeed();
+			siteID = response.posts[ 0 ].site_ID;
+			postID = response.posts[ 0 ].ID;
+		} );
+
+		it( 'Leave a comment on the post', async function () {
+			const response: NewCommentResponse = await restAPIClient.createComment(
+				siteID,
+				postID,
+				comment
+			);
+			commentID = response.ID;
+		} );
+	} );
+
+	describe( `View notification as ${ notificationsUser }`, function () {
+		let page: Page;
+
+		beforeAll( async function () {
+			page = await browser.newPage();
+			const testAccount = new TestAccount( notificationsUser );
+			await testAccount.authenticate( page );
+		} );
+
+		it( 'Open notification using keyboard shortcut', async function () {
+			const navbarComponent = new NavbarComponent( page );
+			await navbarComponent.openNotificationsPanel( { useKeyboard: true } );
+		} );
+
+		it( `See and click notification for the comment left by ${ commentingUser }`, async function () {
+			notificationsComponent = new NotificationsComponent( page );
+			await notificationsComponent.clickNotification( comment );
+		} );
+	} );
+
+	describe( 'Approve comment from notification', function () {
+		it( 'Approve comment', async function () {
+			await notificationsComponent.clickNotificationAction( 'Approve' );
+		} );
+
+		it( 'Unapprove comment', async function () {
+			await notificationsComponent.clickNotificationAction( 'Approve' );
+		} );
+	} );
+
+	describe( 'Like comment from notification', function () {
+		it( 'Like comment', async function () {
+			await notificationsComponent.clickNotificationAction( 'Like' );
+		} );
+
+		it( 'Unlike comment', async function () {
+			await notificationsComponent.clickNotificationAction( 'Liked' );
+		} );
+	} );
+
+	describe( 'Mark comment as spam from notification', function () {
+		it( 'Mark comment as spam', async function () {
+			await notificationsComponent.clickNotificationAction( 'Spam' );
+		} );
+
+		it( 'Undo mark comment as spam', async function () {
+			await notificationsComponent.clickUndo();
+		} );
+	} );
+
+	describe( 'Trash comment from notification', function () {
+		it( 'Trash comment', async function () {
+			await notificationsComponent.clickNotificationAction( 'Trash' );
+		} );
+
+		it( 'Undo trash comment', async function () {
+			await notificationsComponent.clickUndo();
+		} );
+	} );
+
+	// Delete test comment that was created.
+	afterAll( async function () {
```

Comment:
**style:non-blocking**: Similar to above, I'd say this should be moved out to the top level `afterAll` for styling consistency between specs.

Reply:
@worldomonation Thanks for the suggestion! Do you mind clarifying this a bit, please? The current `afterAll` is under the `describe( Some test suite )` level. Is there a level higher than that or do you mean moving it to a higher position in the file (still under the test suite `describe`)?

Reply:
Ah is it already at the level within the top level `describe`? The indentation made it seem like it was nested inside a second-level `describe`. Sorry about that, ignore the comment in that case.

Reply:
No worries! I just wanted to double check. Thanks for going through everything.